cmd/snap-bootstrap/partition/sfdisk.go

```diff
@@ -277,6 +277,21 @@ func deviceLayoutFromPartitionTable(ptable sfdiskPartitionTable) (*DeviceLayout,
 			Type:       p.Type,
 			Filesystem: bd.FSType,
 		}
+
+		// sometimes sfdisk reports what is "0C" in gadget.yaml as "c", so fix
+		// that up now
+		// TODO:UC20: what's a better way?
+		if structure[i].Type == "c" {
```

Comment:
If this needs to be normalized it would be better to normalize any MBR type to the %02X format instead of just 0C.

Reply:
IIRC the parsers for gadget.yaml in `snapd/gadget` and in `ubuntu-image` expect `[0-9A-F]{2}`, so normalizing sounds like a good idea.

Reply:
Added now.